zaza/openstack/charm_tests/cinder_netapp/tests.py

```diff
@@ -55,23 +65,30 @@ def test_cinder_config(self):
             expected_contents,
             timeout=2)
 
+    def _create_volumes(self, n):
+        """Create N volumes."""
+        volumes = []
+        for _ in range(n):
+            name = "zaza{}".format(uuid.uuid1().fields[0])
+            vol = self.cinder_client.volumes.create(
+                name=name,
+                size='1')
+            vol.reset_state('available')
+            volumes.append(vol)
+        self.zaza_volumes.extend(volumes)
+        return volumes
+
     def test_create_volume(self):
-        """Test creating a volume with basic configuration."""
-        test_vol_name = "zaza{}".format(uuid.uuid1().fields[0])
-        vol_new = self.cinder_client.volumes.create(
-            name=test_vol_name,
-            size='1')
-        try:
+        """Test creating volumes with basic configuration."""
+        for vol in self._create_volumes(2):
+            self.assertIsNotNone(vol)
             openstack_utils.resource_reaches_status(
                 self.cinder_client.volumes,
-                vol_new.id,
+                vol.id,
                 wait_iteration_max_time=12000,
                 stop_after_attempt=5,
                 expected_status='available',
                 msg='Volume status wait')
-            test_vol = self.cinder_client.volumes.find(name=test_vol_name)
-            self.assertEqual(
-                getattr(test_vol, 'os-vol-host-attr:host').split('#')[0],
-                'cinder@cinder-netapp')
-        finally:
-            self.cinder_client.volumes.delete(vol_new)
+            test_vol = self.cinder_client.volumes.find(name=vol.name)
+            host = getattr(test_vol, 'os-vol-host-attr:host').split('#')[0]
```

Comment:
This seems to fail with `AttributeError: 'NoneType' object has no attribute 'split'`
See https://review.opendev.org/c/openstack/charm-cinder-netapp/+/815130

Reply:
Maybe we can add a debug to that line for now and re-run the gate?

Reply:
> Maybe we can add a debug to that line for now and re-run the gate?

It sounds like the attr doesn't exist.  Should it exist on the version it is failing on?  Is it a race? Is there something else wrong?

Reply:
Fine with me with adding more debugging output here, if hard to reproduce

Reply:
Running with the default configuration, I'm getting:
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cinder/volume/drivers/netapp/dataontap/client/api.py", line 199, in send_http_request
    response = self._opener.open(request)
  File "/usr/lib/python3.8/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/usr/lib/python3.8/urllib/request.py", line 542, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 1383, in http_open
    return self.do_open(http.client.HTTPConnection, req)
  File "/usr/lib/python3.8/urllib/request.py", line 1357, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 113] EHOSTUNREACH>
```

I'm investigating what's going on with the netapp appliance.